code/app/src/main/java/com/kernelcrew/moodapp/ui/MoodComments.java

```diff
@@ -0,0 +1,303 @@
+
+        // Add to DB using CommentProvider
+        commentProvider.insertComment(comment)
+                .addOnSuccessListener(result -> {
+                    searchInput.setText("");
+                    refreshComments();
+                })
+                .addOnFailureListener(error -> {
+                    Toast.makeText(getContext(), "Failed to add comment: " + error.getMessage(), Toast.LENGTH_LONG).show();
+                    Log.e("MoodComments", "Comment: " + comment.logging());
+                    Log.e("MoodComments", "Error adding comment: " + error);
+                });
+    }
+
+    private void fetchComments() {
+        if (moodEventId == null) {
+            return;
+        }
+
+        CommentAdapter commentAdapter = new CommentAdapter();
+        recyclerView.setAdapter(commentAdapter);
+
+        // Query comments for this mood event
+        commentProvider.getCommentsByMoodEventId(moodEventId)
+                .addOnSuccessListener(comments -> {
+                    commentAdapter.setComments(comments);
+                    commentCountText.setText(String.valueOf(comments.size()));
+                })
+                .addOnFailureListener(error -> {
+                    Log.e("MoodComments", "Error fetching comments: " + error);
+                    Toast.makeText(getContext(), "Failed to load comments", Toast.LENGTH_SHORT).show();
+                });
+    }
+
+    /**
+     * Fetches the mood details document from Firestore using the moodEventId.
+     */
+    private void fetchMoodDetails(String moodEventId) {
+        if (moodEventId == null) return;
+        provider.getMoodEvent(moodEventId)
+                .addOnSuccessListener(moodEvent -> {
+                    if (moodEvent != null) {
+                        bindMoodData(moodEvent);
+                    } else {
+                        Toast.makeText(requireContext(), "Mood event not found", Toast.LENGTH_SHORT).show();
+                    }
+                })
+                .addOnFailureListener(e -> {
+                    Toast.makeText(requireContext(), "Failed to load mood details: " + e.getMessage(), Toast.LENGTH_SHORT).show();
+                });
+        fetchComments();
+    }
+
+    /**
+     * Binds MoodEvent data to the UI components.
+     */
+    private void bindMoodData(MoodEvent moodEvent) {
+        tvMoodState.setText(moodEvent.getEmotion().toString());
+        int moodImageRes = getMoodIconResource(moodEvent.getEmotion().toString());
+        imageMoodIcon.setImageResource(moodImageRes);
+
+        SimpleDateFormat sdf = new SimpleDateFormat("EEE ha", Locale.getDefault());
+        String formattedDate = sdf.format(moodEvent.getCreated());
+
+        dayTimeText.setText(formattedDate);
+        usernameText.setText(moodEvent.getUsername());
+    }
+
+    /**
+     * Refreshes the comments list by fetching the latest comments from Firestore
+     */
+    private void refreshComments() {
```

Comment:
It would be best to listen to changes like we do for `MoodEventProvider::addSnapshotListener`.